ReactiveCocoa/Swift/Property.swift

```diff
@@ -83,6 +83,15 @@ public protocol MutablePropertyType: class, PropertyType {
 	var value: Value { get set }
 }
 
+/// The state of a `MutableProperty`.
+private final class MutablePropertyState<Value> {
```

Comment:
It is necessary so that the `MutableProperty.producer` can still access the latest value after the deallocation of the `MutableProperty`. If we let the producer to capture a `Value`, since `Value` is not guaranteed to be a reference type, the producer might end up to have only a snapshot of the value instead of the latest, thus breaking the semantic.